### PR TITLE
Fixing Sparse Merkle Tree implementation

### LINKED ISSUE #365: Large transaction Livelock
Currently, when a large set of concurrent txs are sent to be mined, only a portion are mined leading to the system eventually hitting a livelock. This can be reproduced by sending 300 or more concurrent txs.

Repro tools:
- https://github.com/alicenet/sendTest 
- https://github.com/alicenet/alicenetjs/blob/main/debugging-scripts/bgTransactionRunner.js

### PR DESCRIPTION
**Pull request by kosegor**
## Scope
Our SMT implementation was inserting duplicated keys to the tree in a very specific situation

## Why?
In order to fix the bug

closes #365

### CODE REVIEW

File: application/utxohandler/utxotrie/utxotrie_test.go [hunk 1]
```diff
@@ -173,3 +187,2457 @@ func TestUTXOTrie(t *testing.T) {
...
+	// Height 24, tx 0
+	tx := &objs.Tx{
+		Fee:  uint256.Zero(),
+		Vin:  make([]*objs.TXIn, 1),
+		Vout: make([]*objs.TXOut, 21),
+	}
+
+	tx.Vin[0] = createVinTx(
+		t,
+		"01011b8f4e22353081e3870bde9171a6fd62924f6221fcb4cba1941896214fb210d92855a64346ca14a8bc9b343b649da07dd38dd94c3881ad305f3af91372ef5ea901",
```
Comment (diligentis) [inline application/utxohandler/utxotrie/utxotrie_test.go:212]: Try not to use magic addresses and numbers unless there's literally no other way to accomplish things. If that is the case, documentation should be provided for how those values came to be. If the values are pulled from a live environment they should be stored in a separate JSON file and read in as part of the test along with instructions for how to regenerate that file in the future. `Golden Files` is one of the terms for this, and [here](https://ieftimov.com/posts/testing-in-go-golden-files/) is a blog post with some additional context and some approaches (not the only approach, just one of).

**Review by diligentis — CHANGES_REQUESTED** [review]
Solid investigation! Just a nit on the test cases to make sure they're clear to other developers.

File: application/utxohandler/utxotrie/utxotrie_test.go (hunk 1 above)
Comment (chgorman) [reply application/utxohandler/utxotrie/utxotrie_test.go:212]: Some form of magic numbers are required.
This comes from the fact that a value in the SMT must first be added and then deleted,
and the additional values which are added must be in a particular form.

Comment (diligentis) [reply application/utxohandler/utxotrie/utxotrie_test.go:212]: I'm defining magic numbers as "literal values used in code without explanation of its meaning".

Using constant literal values is totally fine, but they need to be explained. This test as currently written is incomprehensible to anyone without extensive in depth knowledge of the internals of our implementation, making it impossible for new team members to help maintain the system.

**Review by diligentis — APPROVED** [review]
LGTM, thank you!

**Pull request merged**